engines/dragons/dragons.cpp

```diff
@@ -0,0 +1,1452 @@
+				} else if (event.kbd.keycode == Common::KEYCODE_w) {
+					_wKeyDown = false;
+				} else if (event.kbd.keycode == Common::KEYCODE_a) {
+					_aKeyDown = false;
+				} else if (event.kbd.keycode == Common::KEYCODE_s) {
+					_sKeyDown = false;
+				} else if (event.kbd.keycode == Common::KEYCODE_d) {
+					_dKeyDown = false;
+				} else if (event.kbd.keycode == Common::KEYCODE_o) {
+					_oKeyDown = false;
+				} else if (event.kbd.keycode == Common::KEYCODE_p) {
+					_pKeyDown = false;
+				}
+				break;
+			case Common::EVENT_KEYDOWN:
+				if (event.kbd.keycode == Common::KEYCODE_LEFT) {
+					_leftKeyDown = true;
+				} else if (event.kbd.keycode == Common::KEYCODE_RIGHT) {
+					_rightKeyDown = true;
+				} else if (event.kbd.keycode == Common::KEYCODE_w) {
+					_wKeyDown = true;
+				} else if (event.kbd.keycode == Common::KEYCODE_a) {
+					_aKeyDown = true;
+				} else if (event.kbd.keycode == Common::KEYCODE_s) {
+					_sKeyDown = true;
+				} else if (event.kbd.keycode == Common::KEYCODE_d) {
+					_dKeyDown = true;
+				} else if (event.kbd.keycode == Common::KEYCODE_o) {
+					_oKeyDown = true;
+				} else if (event.kbd.keycode == Common::KEYCODE_p) {
+					_pKeyDown = true;
+				} else if (event.kbd.keycode == Common::KEYCODE_TAB) {
+					_debugMode = !_debugMode;
+				}
+				break;
+			default:
+				break;
+		}
+	}
+}
+
+Common::Error DragonsEngine::run() {
+	_screen = new Screen();
+	_bigfileArchive = new BigfileArchive("bigfile.dat", Common::EN_ANY);
+	_talk = new Talk(this, _bigfileArchive);
+	_dragonFLG = new DragonFLG(_bigfileArchive);
+	_dragonIMG = new DragonIMG(_bigfileArchive);
+	_dragonOBD = new DragonOBD(_bigfileArchive);
+	_dragonRMS = new DragonRMS(_bigfileArchive, _dragonOBD);
+	_dragonVAR = new DragonVAR(_bigfileArchive);
+	_dragonINIResource = new DragonINIResource(_bigfileArchive);
+	_fontManager = new FontManager(this, _screen, _bigfileArchive);
+	_credits = new Credits(this, _fontManager, _bigfileArchive);
+	ActorResourceLoader *actorResourceLoader = new ActorResourceLoader(_bigfileArchive);
+	_actorManager = new ActorManager(actorResourceLoader);
+	_scriptOpcodes = new ScriptOpcodes(this, _dragonFLG);
+	_backgroundResourceLoader = new BackgroundResourceLoader(_bigfileArchive, _dragonRMS);
+	_scene = new Scene(this, _screen, _scriptOpcodes, _actorManager, _dragonRMS, _dragonINIResource, _backgroundResourceLoader);
+
+	_sound = new SoundManager(this, _bigfileArchive, _dragonRMS);
+
+	if (ConfMan.hasKey("save_slot")) {
+		loadGameState(ConfMan.getInt("save_slot"));
+	} else {
+		loadScene(0);
+	}
+
+	_scene->draw();
+	_screen->updateScreen();
+
+	gameLoop();
+
+	delete _scene;
+	delete _actorManager;
+	delete _backgroundResourceLoader;
+	delete _dragonFLG;
+	delete _dragonIMG;
+	delete _dragonRMS;
+	delete _dragonVAR;
+	delete _fontManager;
+	delete _bigfileArchive;
+	delete _screen;
+	delete _sound;
+
+	debug("Ok");
+	return Common::kNoError;
+}
+
+uint16 DragonsEngine::ipt_img_file_related()
+{
```

Comment:
Wrong code formatting, and suddenly you went away from the camel case for the function name. Maybe worth renaming as it looks alien.